src/lib/server.ts

```diff
@@ -88,60 +106,71 @@ export type HttpResponse<Token> = {
     streaming_strategy: Opt<StreamingStrategy<Token>>;
 };
 
-export const HttpRequest = Record({
-    method: text,
-    url: text,
-    headers: Vec(HeaderField),
-    body: blob,
-    certificate_version: Opt(nat16)
-});
-export type HttpRequest = typeof HttpRequest.tsType;
+let nodeServer: NodeServer | undefined = undefined;
 
-export const HttpUpdateRequest = Record({
-    method: text,
-    url: text,
-    headers: Vec(HeaderField),
-    body: blob
-});
-export type HttpUpdateRequest = typeof HttpUpdateRequest.tsType;
+type ServerCallback = () => NodeServer | Promise<NodeServer>;
 
-let server: NodeServer;
+export function serverInit(serverCallback: ServerCallback) {
+    return init([], async () => {
+        nodeServer = await serverCallback();
+    });
+}
 
-export function Server(serverInit: () => NodeServer | Promise<NodeServer>) {
+export function serverPostUpgrade(serverCallback: ServerCallback) {
+    return postUpgrade([], async () => {
+        nodeServer = await serverCallback();
+    });
+}
+
+export const serverHttpRequest = query(
+    [HttpRequest],
+    Manual(HttpResponse()),
+    async (httpRequest) => {
+        await httpHandler(httpRequest, true);
+    },
+    {
+        manual: true
+    }
+);
+
+export const serverHttpRequestUpdate = update(
+    [HttpUpdateRequest],
+    Manual(HttpResponse()),
+    async (httpRequest) => {
+        await httpHandler(httpRequest, false);
+    },
+    {
+        manual: true
+    }
+);
+
+export function serverCanisterMethods(serverCallback: ServerCallback) {
+    return {
+        init: serverInit(serverCallback),
+        postUpgrade: serverPostUpgrade(serverCallback),
+        http_request: serverHttpRequest,
```

Comment:
Why are these snake_case but postUpgrade is camelCase?

Reply:
Because the actual canister method names are used to find http_request and http_request_update, whereas a Rust macro is used for init and postUpgrade, so we don't need to use any specific names